### PR TITLE
Update the release GitHub page from krel

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

This pull request integrates the new announce code that updates the GitHub page to `krel release`. 

The github_page.go code in the announce pkg has been modified to avoid it failing in mock mode, when the release tag does not exist.

#### Which issue(s) this PR fixes:
Part of #1673 

#### Special notes for your reviewer:

The current output creates the release page using the default template. The output from the sample run is in the log and can be seen [rendered in my test repository](https://github.com/puerco/lab/blob/master/ghpage/v1.20.0-beta.3.md). 

Modifying the generated page is now easy (we just need a new template) and is not the focus of this PR.

```release-note
- UpdateGitHubPage() is now implemented in `krel release`.
- The announce package now renders the GitHub page from the template before performing checks in the repository to be able to run while the release tag has not been created.
```

### CODE REVIEW

**Comment by puerco** [comment]
#### Test Run
:heavy_check_mark:  Sample release run: https://console.cloud.google.com/cloud-build/builds/a611426e-a163-4679-acb0-a2fcf596b87f?authuser=1&project=kubernetes-release-test

The github page step begins at line 1101

**Review by saschagrunert — APPROVED** [review]
/lgtm

File: pkg/anago/release.go [hunk 1]
```diff
@@ -445,4 +454,50 @@ func (d *DefaultRelease) CreateAnnouncement() error {
...
+	// Build the options set for the GitHub page
+	ghPageOpts := &announce.GitHubPageOptions{
+		AssetFiles:            assetList,
+		Tag:                   d.state.versions.Prime(),
+		NoMock:                d.options.NoMock,
+		UpdateIfReleaseExists: true,
+		Name:                  "Kubernetes " + d.state.versions.Prime(),
+		Draft:                 false,
+		Owner:                 "kubernetes",
+		Repo:                  "kubernetes",
```
Comment (saschagrunert) [inline pkg/anago/release.go:488]:
```suggestion
		Owner:                 git.DefaultGithubOrg,
		Repo:                  git.DefaultGithubRepo,
```

**Review by saschagrunert — COMMENTED** [review]
Just left my two nits since this needs now a rebase. :point_down:

File: pkg/anago/release.go (hunk 1 above)
Comment (puerco) [reply pkg/anago/release.go]: :white_check_mark: done.

Do you think that later we should have these values in anago or maybe in release?

While I was considering using them yesterday it seemed to me that while having them in git seems like a good failsafe/sane default, if we see the git package in as more general purpose tool perhaps the release process should have it's own constants, decoupled from the git pkg default.

Comment (puerco) [reply pkg/anago/release.go]: I think I did the same thing in the release notes source btw ;) 
I'll fix it at some point.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by xmudrii — APPROVED** [review]
/lgtm

File: pkg/anago/release.go [hunk 2]
```diff
@@ -451,6 +460,52 @@ func (d *DefaultRelease) CreateAnnouncement() error {
...
+	ghPageOpts := &announce.GitHubPageOptions{
+		AssetFiles:            assetList,
+		Tag:                   d.state.versions.Prime(),
+		NoMock:                d.options.NoMock,
+		UpdateIfReleaseExists: true,
+		Name:                  "Kubernetes " + d.state.versions.Prime(),
+		Draft:                 false,
+		Owner:                 git.DefaultGithubOrg,
+		Repo:                  git.DefaultGithubRepo,
+		// PageTemplate: ,     // If we use a custom template, define it here
```
Comment (hasheddan) [inline pkg/anago/release.go:495]: Do we mean to leave this comment in?

Comment (puerco) [reply pkg/anago/release.go:495]: Oh yes I left it intentionally as a guide when we decide what should become of the page  ([ref this in slack](https://kubernetes.slack.com/archives/CJH2GBF7Y/p1606280899195000?thread_ts=1606254574.186400&cid=CJH2GBF7Y))
But I can take it out if it's better :)

**Pull request merged**